packages/env/src/env.tsx

```diff
@@ -57,7 +58,9 @@ export function EnvironmentProvider(props: EnvironmentProviderProps) {
       {mounted && (
         <span
           ref={(el) => {
-            if (el) setNode(el)
+            startTransition(() => {
```

Comment:
I thought this `startTransition` was unnecessary in the previous [PR](https://github.com/chakra-ui/chakra-ui/pull/6171#discussion_r903752912), but I reverted it because `EnvironmentProvider` needed `startTransition`.